### DISCUSSION TITLE
Check if table exists without throwin exception/error

### DISCUSSION THREAD

**Discussion by alannaidon** [adonisjs/core#1678] [Help]
I need to verify if a table exists or not. Is there a way to do it  without throwing error/exception?  

```
async tableExists (tableName) {
   const hasTable = await Database.raw(`SHOW TABLES FROM database2 LIKE '${tableName}'`)
   if (hasTable[0][0]) {
      return true
    } else {
      return false
    }
}  
```  

Also tried:

```
const hasTable = await Database.raw(`SELECT *
      FROM information_schema.tables
      WHERE table_schema = DATABASE2
      AND table_name = ${tableName}`)
```   

Both examples above throws error/exception when the table does not exists and doesn't finish the function.  
I don't want to use a `Try/Catch` and throw an error to check it. Sounds wrong...

**Comment by thetutlage** [comment]
Which version of `@adonisjs/lucid` are you using?

**Reply by alannaidon** [reply]
I'm using Version `4.1.0`

**Reply by thetutlage** [reply]
I believe you can use `await Database.schema.hasTable('tableName')` method
